app/models/user.rb

```diff
@@ -4,21 +4,24 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :pending_purchases
+  has_many :line_items
+  has_one :cart
+
+  after_save :create_cart
+
+  def create_cart
+    Cart.create!(user: self)
+  end
 
   def add_to_cart(book)
```

Comment:
User shouldn't have that association - I removed it. 

My thinking here was that the user is actually taking the action, so the action should be on the user. But I see your thinking, I will move the logic to the cart.